File: backend/src/pbam/interfaces/api/v1/schemas/document.py
"""Pydantic v2 schemas for document/OCR endpoints."""
from datetime import datetime
from decimal import Decimal
from typing import Any
from uuid import UUID

from pydantic import BaseModel


class OcrJobResponse(BaseModel):
    id: UUID
    original_name: str
    file_size_bytes: int
    status: str
    error_message: str | None
    started_at: datetime | None
    completed_at: datetime | None
    committed_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class StagingRowResponse(BaseModel):
    id: UUID
    ocr_job_id: UUID
    sort_order: int
    review_status: str
    account_id: UUID | None
    category_id: UUID | None
    amount_thb: Decimal | None
    original_amount: Decimal | None
    original_currency: str | None
    transaction_type: str | None
    payment_method: str | None
    description: str | None
    transaction_date: str | None
    tags: list[str]
    confidence: dict[str, Any]
    raw_text: str | None

    model_config = {"from_attributes": True}


class StagingRowUpdate(BaseModel):
    account_id: UUID | None = None
    category_id: UUID | None = None
    amount_thb: Decimal | None = None
    original_amount: Decimal | None = None
    original_currency: str | None = None
    exchange_rate: Decimal | None = None
    transaction_type: str | None = None
    payment_method: str | None = None
    description: str | None = None
    transaction_date: str | None = None
    tags: list[str] | None = None


class CommitRequest(BaseModel):
    default_account_id: UUID


class CommitResponse(BaseModel):
    committed_count: int
    job_id: UUID

File: frontend/src/domain/document.ts
/** TypeScript interfaces for the document/OCR domain. */

export type OcrJobStatus = 'pending' | 'processing' | 'review' | 'committed' | 'failed'
export type StagingReviewStatus = 'pending' | 'edited' | 'confirmed' | 'discarded'

export interface OcrJob {
  id: string
  original_name: string
  file_size_bytes: number
  status: OcrJobStatus
  error_message: string | null
  started_at: string | null
  completed_at: string | null
  committed_at: string | null
  created_at: string
}

export interface StagingRow {
  id: string
  ocr_job_id: string
  sort_order: number
  review_status: StagingReviewStatus
  account_id: string | null
  category_id: string | null
  amount_thb: number | null
  original_amount: number | null
  original_currency: string | null
  payment_method: string | null
  transaction_type: string | null
  description: string | null
  transaction_date: string | null
  tags: string[]
  /** Per-field confidence 0.0–1.0 */
  confidence: Record<string, number>
  raw_text: string | null
}
